### PR TITLE
support string arrays in defaultres

### PR DESCRIPTION
**Pull request by dodgex**
see https://github.com/excilys/androidannotations/issues/1625

### CODE REVIEW

File: AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/SharedPrefHandler.java [hunk 1]
```diff
@@ -322,10 +322,21 @@ private IJExpression extractResValue(SharedPrefHolder holder, Element method, IR
...
 			break;
 		}
-		return holder.getContextField().invoke("getResources").invoke(resourceGetMethodName).arg(idRef);
+
+		JInvocation resourceInvocation = holder.getContextField().invoke("getResources").invoke(resourceGetMethodName).arg(idRef);
+
+		if (IRClass.Res.ARRAY.equals(res)) {
+			JInvocation asList = getClasses().ARRAYS.staticInvoke("asList");
+			JInvocation newHashMap = JExpr._new(getClasses().HASH_SET.narrow(getClasses().STRING));
+			resourceInvocation = newHashMap.arg(asList.arg(resourceInvocation));
```
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/SharedPrefHandler.java]: We should handle the empty array here as well (I think it is possible to define an empty string array resource or not?)

**Comment by WonderCsabo** [comment]
Can you extend @DefaultRes javadoc and explain which method return value should use which resource type?

File: AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/SharedPrefHandler.java (hunk 1 above)
Comment (dodgex) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/SharedPrefHandler.java]: emtpy string array resource should not be a prolem. this code generates `new HashSet<>(Arrays.asList(res.getStringArray(R.array.xxx)));` if the resource is empty, the array is emtpy, the list is empty and the set is empty. also i think we have no way to access the number of items of a resource on compile time as the value might be diffrent for e.g. locales or screen sizes.

Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/SharedPrefHandler.java]: Okay.

**Comment by dodgex** [comment]
@WonderCsabo is the table ok?

**Comment by WonderCsabo** [comment]
I think you are missing `long` in the table.

**Comment by dodgex** [comment]
yeah you are right.

**Comment by WonderCsabo** [comment]
Nice, thanks.

**Pull request merged**